iFixit/checkout-ui-extensions
# Samsung part customer attestation

Closes https://github.com/iFixit/checkout-ui-extensions/issues/27
Original spec is here https://github.com/iFixit/ifixit/issues/43322

This should only show up on the US store when the app is eventually deployed to production. Therefore nothing is localized and all text is in English.

I discovered that we can't use `useAppMetafields` to get a metafield for all products in the cart due to a limitation in the hooks Shopify provides. More details in https://github.com/Shopify/ui-extensions/issues/1780. In this PR we use the Storefront API to work around the limitation.

## QA

Make sure the OEM partnership attestation renders on the first page of checkout only when samsung genuine parts are in the cart (as determined by the "OEM Partnership JSON" metafield). [Here](https://ifixit-checkout-dev.myshopify.com/products/samsung-galaxy-s22-ultra-usa-screen-and-battery-genuine) is an example product.

Customers should not be able to complete checkout without checking this box.

In shopify-v2, this component is a popup. It was easier to implement this as a checkbox in this framework, not sure if popups are even possible.

| Screenshot |
| --- |
| ![image](https://github.com/iFixit/checkout-ui-extensions/assets/52104630/61cd0a42-e24a-48b9-a000-ffbb0ce8191e) |
